File: Cpp/7_Reverse_Integer.cpp
class Solution {
public:
    int reverse(int x) {
        long result = x % 10;
        int current = x / 10;
        
        while(current) {
            result = result * 10 + current % 10;
            current /= 10;
        }

        if (result <= INT_MIN || result >= INT_MAX) return 0;
        return result;
    }
};

File: Kotlin/7_Reverse_Integer.kt
class Solution {
    fun reverse(x: Int): Int {
        var result = x % 10L
        var current = x / 10
        
        while(current != 0) {
            result = result * 10 + current % 10
            current /= 10
        }
        
        if (result <= Integer.MIN_VALUE || result >= Integer.MAX_VALUE) return 0
        
        return result.toInt()
    }
}